ko/sql/dblink.rst

```diff
@@ -272,122 +220,370 @@ cubrid_gateway.conf 파일
+Oracle Database에 연결을 하기 위해서는 연결정보를 가지고 있는 tnsnames.ora 파일을 수정해야 한다.
+아래의 기본 형식에 HOST, PORT, SERVICE_NAME 이 세 항목에 연결정보를 작성해야 한다.
+연결정보를 작성한 tnsnames.ora 파일은 TNS_ADMIN 환경변수에서 디렉토리 경로를 설정해야 한다. 
+TNS_ADMIN설정 방법은 "TNS_ADMIN 환경변수 설정" 참고한다.
+
+
+tnsnames.ora 파일의 기본 형식
+
+::
+	
+	net_service_name =
+	  (DESCRIPTION=
+		(ADDRESS = (PROTOCOL = TCP)(HOST = xxx.xxx.xxx.xxx)(PORT = 1521)
+	  )
+	  (CONNECT_DATA =
+		(SERVICE_NAME=service_name)
+	  )
+	)
+
+
+* net_service_name: 데이터베이스 연결을 위한 네트 서비스 이름이며, connection url의 db_name에 사용하는 이름이다.
+* HOST: 데이터베이스에 연결하려는 IP 주소 또는 서버 이름이다.
+* PORT: 연결에 필요한 포트이다. 대부분의 경우 기본 포트는 1521이다.
+* service_name: 연결하려는 데이터베이스의 이름이다.
+
+
+.. note::
+    
+        참고로, net_service_name 이 중복으로 작성이 되어도 에러가 발생되지 않는다. 하지만 중복된 다른 서버에 연결될 수 있으므로, net_service_name 이 반드시 중복되지 않게 설정해야 한다.
+
+
+.. _tns_admin-info:
+
+**TNS_ADMIN 환경변수 설정**
+
+TNS_ADMIN는 tnsnames.ora 파일이 있는 디렉토리 경로를 가리킨다.
+만약 /home/user/myconfigs 에 tnsnames.ora 파일이 있다면 아래와 같이 설정 할 수 있다.
+
+::
+	
+	export TNS_ADMIN=/home/user/myconfigs
+
 
-동일기종의 Brokers와 이기종의 GATEWAY 설정을 했다면, 데이터베이스의 정보를 조회하기 위한 Query문 작성 방법에 대해서 알아본다.
 
-데이터 조회를 위한 DBLINK Query문 작성 방법 두가지가 있다.
+**Oracle을 위한 cubrid_gataway.conf 설정**
+
+게이트웨이에서 oracle에 연결하기 위해서는 아래와 같이 몇 가지 설정이 필요하다.
+
+자세한 :ref:`게이트웨이 설정 파일 <gatewayconf-info>`\ 을 참고한다.
+
+게이트웨이는 oracle에 연결하기 위해서 tnsnames.ora 의 정보를 이용하기 때문에  CGW_LINK_SERVER_IP, CGW_LINK_SERVER_PORT 는 작성하지 않아도 된다. 해당 정보를 작성한 경우에도 게이트웨이는 참조하지 않는다.
 
-첫째, FROM절에 DBLINK 구문을 작성하여 타 데이터베이스의 정보를 조회하는 방법
-아래의 Query문은 IP 192.168.0.1의 타 데이터베이스의 remote_t 테이블 정보를 조회하는 Query문이다.
 
 ::
     
-	SELECT * FROM DBLINK ('192.168.0.1:53000:demodb:user:password:','SELECT col1, col2 FROM remote_t') AS t(col1 int, col2 varchar(32));
+	APPL_SERVER              	=CAS_CGW
+			.
+			.
+			.
+	CGW_LINK_SERVER		        =ORACLE
+	CGW_LINK_SERVER_IP      	=localhost
+	CGW_LINK_SERVER_PORT    	=1521
+	CGW_LINK_ODBC_DRIVER_NAME   =Oracle 12c ODBC driver
+	CGW_LINK_CONNECT_URL_PROPERTY =
+
 
+DBLink를 위한 MySQL 설정
+=======================================
 
-두번째, 위의 DBLINK Query를 보면 타 데이터베이스에 접속하기 위한 정보는 가장 기본적인 정보다. 그래서 Query를 작성할 때 마다 매번 작성해야 하는 번거로움과 사용자 정보(id, password) 가 외부로 노출될 우려가 있다.
-이런 번거로움과 정보 보호를 위해 CREATE SERVER문을 이용하면, Query문 보다 간단하고, 사용자 정보 보호에 도움이 된다.
+MySQL 환경설정
+-------------------------
+ 
+**MySQL ODBC Driver 설치**
 
+게이트웨이에서 MySQL 연결을 하기위해서는 MySQL ODBC Driver가 필요 하다.
+아래의 내용은 MYySQL ODBC Drvier 설치 방법이다.
 
+MySQL Yum 저장소 를 사용하여 Connector/ODBC RPM 패키지를 제공합니다. 시스템의 리포지토리 목록에 MySQL Yum 저장소가 있어야 하며,
+없는경우 MySQL Yum 저장소 다운로드 페이지( https://dev.mysql.com/downloads/repo/yum/ ) 에서 플랫폼에 대한 패키지를 선택하고 다운로드한다.
+
+다운로드한 릴리스 패키지를 설치한다.
 
 ::
     
-    CREATE SERVER remote_srv1 ( HOST='192.168.0.1', PORT=53000, DBNAME=demodb, USER=user, PASSWORD='password');
-    SELECT * FROM DBLINK (remote_srv1, 'SELECT col1 FROM remote_t') AS t(col1 int);
+	$ sudo yum install mysql80-community-release-el6-{version-number}.noarch.rpm
+
 
+Yum을 사용하여 저장소를 업데이트한다.
 
+::
+    
+	$ su root
+	$ yum update mysql-community-release

```

Comment:
$ sudo yum update mysql-community-release

Reply:
Thank you for your comment